src/libcore/array.rs

```diff
@@ -210,3 +203,31 @@ macro_rules! array_impl_default {
 }
 
 array_impl_default!{32, T T T T T T T T T T T T T T T T T T T T T T T T T T T T T T T T}
+
+macro_rules! array_impl_clone {
+    {$n:expr, $i:expr, $($idx:expr,)*} => {
+        #[stable(feature = "rust1", since = "1.0.0")]
+        impl<T: Clone> Clone for [T; $n] {
+            fn clone(&self) -> [T; $n] {
+                let temp = [&self[$i], $(&self[$idx]),*];
+                [temp[$i].clone(), $(temp[$idx].clone()),*]
```

Comment:
Good idea, thank you. For some reason I couldn't figure out how to reverse the indices.